python/paddle/fluid/tests/unittests/mkldnn/test_cast_mkldnn_op.py

```diff
@@ -0,0 +1,134 @@
+        self.x_bf16 = convert_float_to_uint16(self.x_fp32)
+
+        self.inputs = {'X': self.x_bf16}
+        self.outputs = {'Out': self.x_fp32}
+        self.attrs = {
+            'in_dtype': int(core.VarDesc.VarType.BF16),
+            'out_dtype': int(core.VarDesc.VarType.FP32),
+            'use_mkldnn': True
+        }
+        self.op_type = 'cast'
+
+    def test_check_output(self):
+        self.check_output(check_dygraph=False)
+
+    def test_check_grad(self):
+        self.check_grad_with_place(
+            core.CPUPlace(), ["X"],
+            "Out",
+            check_dygraph=False,
+            user_defined_grads=[self.x_bf16],
+            user_defined_grad_outputs=[self.x_fp32])
+
+
+class TestCastFP32ToBF16MKLDNNOp(OpTest):
+    def setUp(self):
+        self.x_fp32 = np.random.random(size=[2, 6]).astype("float32")
+        self.x_bf16 = convert_float_to_uint16(self.x_fp32)
+
+        self.inputs = {'X': self.x_fp32}
+        self.outputs = {'Out': self.x_bf16}
+        self.attrs = {
+            'in_dtype': int(core.VarDesc.VarType.FP32),
+            'out_dtype': int(core.VarDesc.VarType.BF16),
+            'use_mkldnn': True
+        }
+        self.op_type = 'cast'
+
+    def test_check_output(self):
+        self.check_output(check_dygraph=False)
+
+    def test_check_grad(self):
+        self.check_grad_with_place(
+            core.CPUPlace(), ["X"],
+            "Out",
+            check_dygraph=False,
+            user_defined_grads=[self.x_fp32],
+            user_defined_grad_outputs=[self.x_bf16])
+
+
+class TestCastBF16ToBF16MKLDNNOp(OpTest):
+    def setUp(self):
+        self.x_fp32 = np.random.random(size=[6, 13]).astype("float32")
+        self.x_bf16 = convert_float_to_uint16(self.x_fp32)
+
+        self.inputs = {'X': self.x_bf16}
+        self.outputs = {'Out': self.x_bf16}
+        self.attrs = {
+            'in_dtype': int(core.VarDesc.VarType.BF16),
+            'out_dtype': int(core.VarDesc.VarType.BF16),
+            'use_mkldnn': True
+        }
+        self.op_type = 'cast'
+
+    def test_check_output(self):
+        self.check_output(check_dygraph=False)
+
+    def test_check_grad(self):
+        self.check_grad_with_place(
+            core.CPUPlace(), ["X"],
+            "Out",
+            check_dygraph=False,
+            user_defined_grads=[self.x_bf16],
+            user_defined_grad_outputs=[self.x_bf16])
+
+
+class TestCastFP32ToFP32MKLDNNOp(OpTest):
+    def setUp(self):
+        self.x_fp32 = np.random.random(size=[7, 15]).astype("float32")
+
+        self.inputs = {'X': self.x_fp32}
+        self.outputs = {'Out': self.x_fp32}
+        self.attrs = {
+            'in_dtype': int(core.VarDesc.VarType.FP32),
+            'out_dtype': int(core.VarDesc.VarType.FP32),
+            'use_mkldnn': True
+        }
+        self.op_type = 'cast'
+
+    def test_check_output(self):
+        self.check_output(check_dygraph=False)
+
+    def test_check_grad(self):
```

Comment:
Maybe you could unify in one common class code of test_check_grad, test_check_output so to not repeat that. With proper naming the only changing part would be setting input, output and attributes of the op.

Reply:
Another nice idea, done!